### PR TITLE
Gauge card

### PR DESCRIPTION
**Pull request by davidicus**
Closes ##849

**Summary**

Implementation of Gauge card component 

**Change List (commits, features, bugs, etc)**

- create GaugeCard.jsx, _gauge-card.scss GaugeCard.story.jsx, GaugeCard.test.jsx
- updated CardRender.jsx and Dashboard.story.jsx to include GaugeCard

**Acceptance Test (how to verify the PR)**

- Go to GaugeCard story and go to Dashboard story and make sure it renders properly

### CODE REVIEW

File: src/components/GaugeCard/GaugeCard.jsx [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+
+export const getColor = (gauge, value) => {
+  let { color } = gauge;
+  let grade = '';
+  const comparisons = {
+    '<': (a, b) => a < b,
+    '>': (a, b) => a > b,
+    '=': (a, b) => a === b,
+    '<=': (a, b) => a <= b,
+    '>=': (a, b) => a >= b,
```
Comment (stuckless) [inline src/components/GaugeCard/GaugeCard.jsx]: Imagine how much more powerful the comparisons could be if we were allowed to supply our own custom comparsion functions.  Doing ranges would be breeze.

File: src/components/GaugeCard/GaugeCard.jsx [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+  };
+  if (gauge.thresholds) {
+    gauge.thresholds.forEach(thresh => {
```
Comment (stuckless) [inline src/components/GaugeCard/GaugeCard.jsx]: do we need to iterate all of them?  could we use something like `every()` instead?

File: src/components/GaugeCard/GaugeCard.jsx [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+  values,
+  data,
+  isLoading,
+  hasMoreData,
+  size,
+  className,
+  ...others
+}) => {
+  const [loadedState, setLoadedState] = useState(false);
+  useEffect(() => {
```
Comment (stuckless) [inline src/components/GaugeCard/GaugeCard.jsx]: I'm confused by this.  What are we doing here?  Seems like we are setting our loaded state to true after 1 second??

File: src/components/GaugeCard/GaugeCard.jsx [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+  isLoading: false,
+  description: null,
+  content: {
+    gauges: [
+      {
+        dataSourceId: null,
+        units: '%',
+        minimumValue: 0,
+        maximumValue: 100,
+        renderValueFunction: null,
```
Comment (stuckless) [inline src/components/GaugeCard/GaugeCard.jsx]: I like this... but I didn't see this used anywhere.

File: src/components/GaugeCard/GaugeCard.jsx (hunk 3 above)
Comment (davidicus) [reply src/components/GaugeCard/GaugeCard.jsx]: This is to trigger the load animation of the gauge

File: src/components/GaugeCard/GaugeCard.jsx (hunk 4 above)
Comment (davidicus) [reply src/components/GaugeCard/GaugeCard.jsx]: I wanted to get this simple implementation out to satisfy the current use case. But we have plans to extend this to accept a threshold function, different types of gauges, etc.

File: src/components/GaugeCard/GaugeCard.jsx (hunk 2 above)
Comment (davidicus) [reply src/components/GaugeCard/GaugeCard.jsx]: I believe that would still need to iterate over the entire array, no? I think we need to run each comparison to be sure it doesn't trigger .

File: src/components/GaugeCard/GaugeCard.jsx (hunk 1 above)
Comment (davidicus) [reply src/components/GaugeCard/GaugeCard.jsx]: I will address these changes and comments in our upcoming sprint. [Issue](https://github.com/IBM/carbon-addons-iot-react/issues/888) to track

File: src/components/GaugeCard/GaugeCard.jsx (hunk 4 above)
Comment (davidicus) [reply src/components/GaugeCard/GaugeCard.jsx]: I will address these changes and comments in our upcoming sprint. [Issue](https://github.com/IBM/carbon-addons-iot-react/issues/888) to track

**Comment by davidicus** [comment]
I will address these changes and comments in our upcoming sprint. [Issue](https://github.com/IBM/carbon-addons-iot-react/issues/888) to track

**Pull request merged**